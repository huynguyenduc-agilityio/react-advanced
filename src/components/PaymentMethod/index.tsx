import { ElementType, forwardRef } from 'react';
import {
  Flex,
  Icon,
  Text,
  Radio,
  RadioGroup,
  Stack,
  Box,
} from '@chakra-ui/react';

export interface Option {
  title: string;
  content: string;
  value: string;
  icon: ElementType;
}

export interface PaymentMethodProps {
  options: Option[];
  value?: string;
  name?: string;
  onChange?: (value: string) => void;
}

const PaymentMethod = forwardRef<HTMLInputElement, PaymentMethodProps>(
  ({ options, value = options[0]?.value, onChange, name, ...rest }, ref) => {
    return (
      <RadioGroup
        w="full"
        name={name}
        value={value}
        onChange={onChange}
        {...rest}
      >
        <Stack>
          {options.map(({ title, content, icon, value: optionValue }) => (
            <Box
              key={optionValue}
              borderRadius="lg"
              bgColor={value === optionValue ? '#1A2561' : 'midNightBlue'}
              border="0.6px solid"
              borderColor={value === optionValue ? '#4b52DF' : 'slateBlue'}
            >
              <Radio ref={ref} value={optionValue} p={4} w="full">
                <Flex align="center" gap={2}>
                  <Icon as={icon} />
                  <Flex direction="column">
                    <Text fontSize="xs" color="white">
                      {title}
                    </Text>
                    <Text fontSize="xs" color="pastelBlue">
                      {content}
                    </Text>
                  </Flex>
                </Flex>
              </Radio>
            </Box>
          ))}
        </Stack>
      </RadioGroup>
    );
  },
);

export default PaymentMethod;
