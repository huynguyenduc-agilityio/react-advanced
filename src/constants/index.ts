export * from './message';
export * from './image';
export * from './regex';
export * from './form';
export * from './file';
export * from './payment';
export * from './config';
export * from './router';
export * from './pagination';
export * from './httpsRequest';
export * from './api';
export * from './queries';
