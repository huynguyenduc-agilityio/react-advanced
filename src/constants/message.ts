export const ERROR_MESSAGES = {
  FIELD_REQUIRED: (fieldName: string) => `${fieldName} is required`,
  MAX_SIZE_IMAGE: 'Image size must be less than 3MB',
  INVALID_FILE_TYPE: 'File must be a PNG, JPEG, or JPG',
  EMPTY_DATA: 'No data found',
  INVALID_EMAIL: 'Invalid email address',
  INVALID_PHONE: 'The number entered is not valid',
  DELETE_USER: 'An error occurred while deleting the user',
  ADD_USER: 'An error occurred while adding the user',
  UPDATE_USER: 'An error occurred while updating the user',
  UPLOAD_IMAGE_FAILED: 'Upload image failed',
};

export const SUCCESS_MESSAGES = {
  ADD_USER: 'User added successfully',
  UPDATE_USER: 'User updated successfully',
  DELETE_USER: 'User deleted successfully',
  IMAGE: 'Image uploaded successfully',
};
