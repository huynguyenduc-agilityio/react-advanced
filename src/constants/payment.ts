import {
  AmericanExpressIcon,
  MasterCardIcon,
  VisaIcon,
} from '@/components/Icons';

export const PAYMENT_METHODS = [
  {
    title: 'VISA **** 8092',
    content: 'Expires on 12/26',
    value: 'visa',
    icon: VisaIcon,
  },
  {
    title: 'Mastercard **** 8092',
    content: 'Expires on 12/26',
    value: 'mastercard',
    icon: MasterCardIcon,
  },
  {
    title: 'American Express **** 8092',
    content: 'Expires on 12/26',
    value: 'american-express',
    icon: AmericanExpressIcon,
  },
];
