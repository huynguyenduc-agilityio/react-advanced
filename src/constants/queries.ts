// Type
import { TQueryKey } from '@/types';

export const usersQueryKeys = {
  all: [{ scope: 'users' }] as const,
  lists: () => [{ ...usersQueryKeys.all[0], entity: 'list' }] as const,
  list: ({ page, pageSize, filter }: TQueryKey) =>
    [
      {
        ...usersQueryKeys.lists()[0],
        page,
        pageSize,
        ...(filter && { filter }),
      },
    ] as const,
  details: () => [{ ...usersQueryKeys.all[0], entity: 'detail' }] as const,
  detail: (id?: string) => [{ ...usersQueryKeys.details()[0], id }] as const,
};
