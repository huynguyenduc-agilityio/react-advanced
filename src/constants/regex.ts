export const REGEX = {
  IMG: /\.(svg|png|jpe?g|gif)$/i,
  PHONE: /^\(\d{3}\) \d{3}-\d{4}$/,
  EMAIL: /^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,}$/i,
};
