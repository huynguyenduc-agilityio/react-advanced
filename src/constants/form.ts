// Icons
import { WebIcon } from '@/components/Icons';
import { FaPhoneAlt, FaUser } from 'react-icons/fa';
import { HiMail } from 'react-icons/hi';
import { HiMapPin } from 'react-icons/hi2';
import { IoBagSharp } from 'react-icons/io5';
import { RiHomeFill } from 'react-icons/ri';

// Constants
import { ERROR_MESSAGES } from './message';
import { ACCEPTED_FILE_TYPES, MAX_UPLOAD_SIZE } from './file';

export const VALIDATION_RULES = {
  IMAGE: {
    validate: (file: File) => {
      if (file.size > MAX_UPLOAD_SIZE) {
        return ERROR_MESSAGES.MAX_SIZE_IMAGE;
      }

      if (!ACCEPTED_FILE_TYPES.includes(file.type)) {
        return ERROR_MESSAGES.INVALID_FILE_TYPE;
      }
    },
  },
};

export const BASIC_FORM_FIELD = [
  {
    label: 'Phone',
    key: 'phone',
    name: 'phone',
    isRequired: true,
    ariaLabel: 'phone',
    icon: FaPhoneAlt,
    placeholder: '(123) 456-7890',
  },
  {
    label: 'Position',
    key: 'position',
    name: 'position',
    isRequired: true,
    ariaLabel: 'position',
    icon: IoBagSharp,
    placeholder: 'Please enter position',
  },
  {
    label: 'Location',
    key: 'location',
    name: 'location',
    isRequired: true,
    ariaLabel: 'location',
    icon: HiMapPin,
    placeholder: 'Please enter location',
  },
  {
    label: 'Website',
    key: 'website',
    name: 'website',
    isRequired: true,
    ariaLabel: 'website',
    icon: WebIcon,
    placeholder: 'Please enter website',
  },
  {
    label: 'Company',
    key: 'company',
    name: 'company',
    isRequired: true,
    ariaLabel: 'company',
    icon: IoBagSharp,
    placeholder: 'Please enter company',
  },
];

export const PERSONAL_FORM_FIELD = [
  {
    label: 'Full name',
    key: 'name',
    name: 'name',
    isRequired: true,
    ariaLabel: 'name',
    icon: FaUser,
    placeholder: 'Please enter name',
  },
  {
    label: 'Email address',
    key: 'email',
    name: 'email',
    isRequired: true,
    ariaLabel: 'email',
    icon: HiMail,
    placeholder: 'Please enter email',
  },
];

export const TEAM_FORM_FIELD = [
  {
    label: 'Team Name',
    key: 'teamName',
    name: 'teamName',
    isRequired: true,
    ariaLabel: 'teamName',
    icon: FaPhoneAlt,
    placeholder: 'Please enter team name',
  },
  {
    label: 'Rank',
    key: 'rank',
    name: 'rank',
    isRequired: true,
    ariaLabel: 'rank',
    icon: IoBagSharp,
    placeholder: 'Please enter rank',
    type: 'select',
    options: [
      { key: 'marketing', value: 'marketing', label: 'Marketing' },
      { key: 'design', value: 'design', label: 'Design' },
      { key: 'senior', value: 'senior', label: 'Senior' },
    ],
  },
  {
    label: 'Office',
    key: 'office',
    name: 'office',
    isRequired: true,
    ariaLabel: 'office',
    icon: HiMapPin,
    placeholder: 'Please enter office',
  },
  {
    label: 'Mail',
    key: 'teamMail',
    name: 'teamMail',
    ariaLabel: 'teamMail',
    icon: WebIcon,
    placeholder: 'Please enter mail',
  },
];

export const BILL_FORM_FIELD = [
  {
    label: 'Full name',
    key: 'billName',
    name: 'billName',
    isRequired: true,
    ariaLabel: 'billName',
    icon: FaUser,
    placeholder: 'Please enter full name',
  },
  {
    label: 'Address',
    key: 'billAddress',
    name: 'billAddress',
    isRequired: true,
    ariaLabel: 'billAddress',
    icon: RiHomeFill,
    placeholder: 'Please enter address',
  },
  {
    label: 'State',
    key: 'state',
    name: 'state',
    isRequired: true,
    ariaLabel: 'state',
    icon: HiMapPin,
    placeholder: 'Please enter state',
  },
  {
    label: 'Zip Code',
    key: 'zipCode',
    name: 'zipCode',
    isRequired: true,
    ariaLabel: 'zipCode',
    icon: HiMail,
    placeholder: 'Please enter zip code',
  },
];

export const GENERAL_FORM_FIELD = [
  {
    label: 'I’m mentioned in a message',
    key: 'mentionMessage',
    name: 'mentionMessage',
  },
  {
    label: 'Someone replies to any message',
    key: 'replyMessage',
    name: 'replyMessage',
  },
  {
    label: 'I’m assigned a task',
    key: 'assignTask',
    name: 'assignTask',
  },
  {
    label: 'A task is overdue',
    key: 'taskOverdue',
    name: 'taskOverdue',
  },
];

export const SUMMARY_FORM_FIELD = [
  {
    label: 'Daily summary',
    key: 'dailySummary',
    name: 'dailySummary',
  },
  {
    label: 'Weekly summary',
    key: 'weeklySummary',
    name: 'weeklySummary',
  },
  {
    label: 'Monthly summary',
    key: 'monthlySummary',
    name: 'monthlySummary',
  },
  {
    label: 'Annually summary',
    key: 'annuallySummary',
    name: 'annuallySummary',
  },
];
