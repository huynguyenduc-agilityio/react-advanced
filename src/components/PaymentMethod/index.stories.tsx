import { useState } from 'react';
import { Box, Flex } from '@chakra-ui/react';
import { Meta, StoryObj } from '@storybook/react';

// Constants
import { PAYMENT_METHODS } from '@/constants';

import PaymentMethod, { PaymentMethodProps } from '.';

const options = PAYMENT_METHODS;

const meta: Meta<typeof PaymentMethod> = {
  title: 'Components/PaymentMethod',
  component: PaymentMethod,
  parameters: {
    layout: 'centered',
  },

  tags: ['autodocs'],
};

export default meta;

type Story = StoryObj<typeof PaymentMethod>;

const WithState: React.FC<PaymentMethodProps> = ({
  options,
  value: initialValue,
  ...rest
}) => {
  const [value, setValue] = useState<string>(initialValue || options[0]?.value);

  return (
    <Flex justifyContent="center" w="1000px" p={9} bgColor="midNightBlue">
      <Box w="550px">
        <PaymentMethod
          options={options}
          value={value}
          onChange={setValue}
          {...rest}
        />
      </Box>
    </Flex>
  );
};

export const Default: Story = {
  render: (args) => <WithState {...args} />,
  args: { options, value: 'visa' },
};

export const SelectMasterCard: Story = {
  render: (args) => <WithState {...args} />,
  args: { options, value: 'mastercard' },
};

export const SelectAmericanExpress: Story = {
  render: (args) => <WithState {...args} />,
  args: { options, value: 'american-express' },
};
