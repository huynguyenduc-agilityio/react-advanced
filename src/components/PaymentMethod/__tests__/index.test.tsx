import { fireEvent, render } from '@testing-library/react';

import { PAYMENT_METHODS } from '@/constants';
import PaymentMethod from '..';

describe('PaymentMethod Component', () => {
  it('renders all payment options', () => {
    const { getByText } = render(
      <PaymentMethod options={PAYMENT_METHODS} value="credit" />,
    );

    expect(getByText('VISA **** 8092')).toBeInTheDocument();
    expect(getByText('Mastercard **** 8092')).toBeInTheDocument();
    expect(getByText('American Express **** 8092')).toBeInTheDocument();
  });

  it('allows selecting a different payment method', () => {
    const handleChange = jest.fn();
    const { getByText } = render(
      <PaymentMethod
        options={PAYMENT_METHODS}
        value="credit"
        onChange={handleChange}
      />,
    );

    fireEvent.click(getByText('Mastercard **** 8092'));
    expect(handleChange).toHaveBeenCalledWith('mastercard');
  });

  it('calls onChange when selecting a payment option', () => {
    const handleChange = jest.fn();
    const { getByRole } = render(
      <PaymentMethod
        options={PAYMENT_METHODS}
        value="credit"
        onChange={handleChange}
      />,
    );

    fireEvent.click(getByRole('radio', { name: /Mastercard/i }));
    expect(handleChange).toHaveBeenCalledTimes(1);
  });

  it('should match snapshot', () => {
    const { container } = render(
      <PaymentMethod options={PAYMENT_METHODS} value="credit" />,
    );

    expect(container).toMatchSnapshot();
  });
});
