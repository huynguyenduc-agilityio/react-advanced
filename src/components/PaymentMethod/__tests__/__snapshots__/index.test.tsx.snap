// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaymentMethod Component should match snapshot 1`] = `
<div>
  <div
    class="chakra-radio-group css-13rbvqk"
    role="radiogroup"
  >
    <div
      class="chakra-stack css-8g8ihq"
    >
      <div
        class="css-x960pu"
      >
        <label
          class="chakra-radio css-jx3t55"
          for="radio-:rd:"
        >
          <input
            class="chakra-radio__input"
            id="radio-:rd:"
            name="radio-:rc:"
            style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
            type="radio"
            value="visa"
          />
          <span
            aria-hidden="true"
            class="chakra-radio__control css-ssalds"
          />
          <span
            class="chakra-radio__label css-1y8kf23"
          >
            <div
              class="css-j2at52"
            >
              <svg
                fill="none"
                height="29"
                viewBox="0 0 40 29"
                width="40"
              >
                <rect
                  fill="#CB3CFF"
                  height="28"
                  rx="4"
                  width="39.9089"
                  x="0.0269775"
                  y="0.0703125"
                />
                <path
                  d="M17.635 18.8247H15.1893L16.7191 9.34558H19.1646L17.635 18.8247Z"
                  fill="white"
                />
                <path
                  d="M26.4992 9.57739C26.0168 9.3856 25.2516 9.17383 24.3056 9.17383C21.8904 9.17383 20.1896 10.4645 20.1791 12.3099C20.1591 13.6713 21.3968 14.4275 22.3226 14.8815C23.2688 15.3454 23.5905 15.6482 23.5905 16.0616C23.5808 16.6967 22.8259 16.9894 22.1217 16.9894C21.1453 16.9894 20.6221 16.8385 19.827 16.4852L19.505 16.3337L19.1627 18.4616C19.7363 18.7235 20.7931 18.9559 21.8904 18.9661C24.4566 18.9661 26.1272 17.6953 26.147 15.7288C26.1568 14.6498 25.5032 13.8229 24.0941 13.1473C23.2387 12.7136 22.7149 12.4211 22.7149 11.9773C22.7249 11.5739 23.1579 11.1607 24.1235 11.1607C24.9186 11.1405 25.5028 11.332 25.9454 11.5236L26.1666 11.6243L26.4992 9.57739Z"
                  fill="white"
                />
                <path
                  d="M29.7513 15.4666C29.9527 14.922 30.7278 12.8144 30.7278 12.8144C30.7176 12.8347 30.9288 12.2598 31.0495 11.9069L31.2205 12.7237C31.2205 12.7237 31.6836 14.9927 31.7842 15.4666C31.4019 15.4666 30.2344 15.4666 29.7513 15.4666ZM32.7702 9.34558H30.8785C30.2951 9.34558 29.8519 9.51687 29.6002 10.1321L25.9674 18.8246H28.5336C28.5336 18.8246 28.9561 17.6546 29.0469 17.4027C29.3284 17.4027 31.8248 17.4027 32.187 17.4027C32.2572 17.7355 32.4788 18.8246 32.4788 18.8246H34.7433L32.7702 9.34558Z"
                  fill="white"
                />
                <path
                  d="M13.1464 9.34558L10.7512 15.8095L10.4895 14.4985C10.0467 12.9858 8.65794 11.3424 7.10815 10.5252L9.30203 18.8147H11.8883L15.7325 9.34558H13.1464Z"
                  fill="white"
                />
                <path
                  d="M8.52667 9.34558H4.59177L4.55151 9.53713C7.62099 10.3238 9.65384 12.2202 10.489 14.4995L9.63364 10.1425C9.49281 9.537 9.06003 9.36555 8.52667 9.34558Z"
                  fill="white"
                />
              </svg>
              <div
                class="css-j7qwjs"
              >
                <p
                  class="chakra-text css-1o3wn7d"
                >
                  VISA **** 8092
                </p>
                <p
                  class="chakra-text css-1q8knsm"
                >
                  Expires on 12/26
                </p>
              </div>
            </div>
          </span>
        </label>
      </div>
      <div
        class="css-x960pu"
      >
        <label
          class="chakra-radio css-jx3t55"
          for="radio-:re:"
        >
          <input
            class="chakra-radio__input"
            id="radio-:re:"
            name="radio-:rc:"
            style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
            type="radio"
            value="mastercard"
          />
          <span
            aria-hidden="true"
            class="chakra-radio__control css-ssalds"
          />
          <span
            class="chakra-radio__label css-1y8kf23"
          >
            <div
              class="css-j2at52"
            >
              <svg
                fill="none"
                height="29"
                viewBox="0 0 41 29"
                width="41"
              >
                <rect
                  fill="#0A1330"
                  height="28"
                  rx="4"
                  width="40"
                  x="0.115234"
                  y="0.0703125"
                />
                <path
                  clip-rule="evenodd"
                  d="M20.1139 19.0559C18.9831 20.0316 17.5162 20.6206 15.9134 20.6206C12.3369 20.6206 9.4375 17.688 9.4375 14.0703C9.4375 10.4527 12.3369 7.51999 15.9134 7.51999C17.5162 7.51999 18.9831 8.10899 20.1139 9.08469C21.2447 8.10899 22.7115 7.51999 24.3144 7.51999C27.8909 7.51999 30.7903 10.4527 30.7903 14.0703C30.7903 17.688 27.8909 20.6206 24.3144 20.6206C22.7115 20.6206 21.2447 20.0316 20.1139 19.0559Z"
                  fill="#ED0006"
                  fill-rule="evenodd"
                />
                <path
                  clip-rule="evenodd"
                  d="M20.1163 19.0559C21.5088 17.8545 22.3918 16.0667 22.3918 14.0703C22.3918 12.0739 21.5088 10.2861 20.1163 9.08468C21.2471 8.10899 22.714 7.51999 24.3168 7.51999C27.8934 7.51999 30.7927 10.4527 30.7927 14.0703C30.7927 17.688 27.8934 20.6206 24.3168 20.6206C22.714 20.6206 21.2471 20.0316 20.1163 19.0559Z"
                  fill="#F9A000"
                  fill-rule="evenodd"
                />
                <path
                  clip-rule="evenodd"
                  d="M20.1145 19.056C21.507 17.8545 22.39 16.0667 22.39 14.0703C22.39 12.074 21.507 10.2862 20.1145 9.08472C18.7221 10.2862 17.8391 12.074 17.8391 14.0703C17.8391 16.0667 18.7221 17.8545 20.1145 19.056Z"
                  fill="#FF5E00"
                  fill-rule="evenodd"
                />
              </svg>
              <div
                class="css-j7qwjs"
              >
                <p
                  class="chakra-text css-1o3wn7d"
                >
                  Mastercard **** 8092
                </p>
                <p
                  class="chakra-text css-1q8knsm"
                >
                  Expires on 12/26
                </p>
              </div>
            </div>
          </span>
        </label>
      </div>
      <div
        class="css-x960pu"
      >
        <label
          class="chakra-radio css-jx3t55"
          for="radio-:rf:"
        >
          <input
            class="chakra-radio__input"
            id="radio-:rf:"
            name="radio-:rc:"
            style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
            type="radio"
            value="american-express"
          />
          <span
            aria-hidden="true"
            class="chakra-radio__control css-ssalds"
          />
          <span
            class="chakra-radio__label css-1y8kf23"
          >
            <div
              class="css-j2at52"
            >
              <svg
                fill="none"
                height="29"
                viewBox="0 0 41 29"
                width="41"
              >
                <rect
                  fill="#0A1330"
                  height="28"
                  rx="4"
                  width="40"
                  x="0.115234"
                  y="0.0703125"
                />
                <path
                  d="M9.64197 20.2535V14.9009H15.3101L15.9182 15.6935L16.5464 14.9009H37.1203V19.8844C37.1203 19.8844 36.5822 20.2481 35.9599 20.2535H24.5678L23.8822 19.4097V20.2535H21.6354V18.8132C21.6354 18.8132 21.3285 19.0142 20.6649 19.0142H19.9002V20.2535H16.4984L15.8911 19.4438L15.2745 20.2535H9.64197Z"
                  fill="#016FD0"
                />
                <path
                  d="M3.02356 10.8667L4.30174 7.88713H6.51222L7.2376 9.55615V7.88713H9.98543L10.4172 9.09345L10.8359 7.88713H23.1708V8.49359C23.1708 8.49359 23.8192 7.88713 24.8849 7.88713L28.8871 7.90112L29.6 9.54829V7.88713H31.8995L32.5324 8.8333V7.88713H34.853V13.2398H32.5324L31.9259 12.2905V13.2398H28.5474L28.2076 12.396H27.2993L26.9651 13.2398H24.6739C23.757 13.2398 23.1708 12.6457 23.1708 12.6457V13.2398H19.7162L19.0306 12.396V13.2398H6.18475L5.84523 12.396H4.93986L4.60274 13.2398H3.02356V10.8667Z"
                  fill="#016FD0"
                />
                <path
                  d="M4.75328 8.54683L3.0293 12.5552H4.1517L4.4698 11.7525H6.31904L6.63549 12.5552H7.78262L6.06028 8.54683H4.75328ZM5.39112 9.47969L5.9548 10.8823H4.8258L5.39112 9.47969Z"
                  fill="white"
                />
                <path
                  d="M7.90173 12.5542V8.54588L9.49673 8.55181L10.4244 11.1361L11.3299 8.54588H12.9122V12.5542H11.9101V9.60071L10.8478 12.5542H9.96902L8.90382 9.60071V12.5542H7.90173Z"
                  fill="white"
                />
                <path
                  d="M13.5983 12.5543V8.54588H16.8683V9.4425H14.6109V10.1281H16.8155V10.972H14.6109V11.684H16.8683V12.5543H13.5983Z"
                  fill="white"
                />
                <path
                  d="M17.4473 8.54683V12.5552H18.4494V11.1312H18.8713L20.0728 12.5552H21.2974L19.9789 11.0784C20.52 11.0328 21.0782 10.5683 21.0782 9.84725C21.0782 9.00375 20.4162 8.54683 19.6773 8.54683H17.4473ZM18.4494 9.44344H19.5949C19.8696 9.44344 20.0695 9.65839 20.0695 9.86538C20.0695 10.1317 19.8105 10.2873 19.6097 10.2873H18.4494V9.44344Z"
                  fill="white"
                />
                <path
                  d="M22.5121 12.5543H21.4889V8.54588H22.5121V12.5543Z"
                  fill="white"
                />
                <path
                  d="M24.9378 12.5543H24.7169C23.6483 12.5543 22.9995 11.7124 22.9995 10.5666C22.9995 9.39243 23.6411 8.54588 24.9905 8.54588H26.0981V9.49524H24.95C24.4022 9.49524 24.0148 9.92275 24.0148 10.5764C24.0148 11.3527 24.4578 11.6788 25.096 11.6788H25.3597L24.9378 12.5543Z"
                  fill="white"
                />
                <path
                  d="M27.1189 8.54683L25.3949 12.5552H26.5173L26.8354 11.7525H28.6846L29.0011 12.5552H30.1482L28.4259 8.54683H27.1189ZM27.7567 9.47969L28.3204 10.8823H27.1914L27.7567 9.47969Z"
                  fill="white"
                />
                <path
                  d="M30.2645 12.5543V8.54588H31.5386L33.1653 11.0643V8.54588H34.1674V12.5543H32.9346L31.2666 9.96992V12.5543H30.2645Z"
                  fill="white"
                />
                <path
                  d="M10.3285 19.5685V15.5601H13.5985V16.4567H11.3411V17.1423H13.5457V17.9862H11.3411V18.6982H13.5985V19.5685H10.3285Z"
                  fill="white"
                />
                <path
                  d="M26.351 19.5685V15.5601H29.6209V16.4567H27.3636V17.1423H29.5577V17.9862H27.3636V18.6982H29.6209V19.5685H26.351Z"
                  fill="white"
                />
                <path
                  d="M13.7254 19.5684L15.3175 17.589L13.6875 15.5601H14.95L15.9208 16.8143L16.8948 15.5601H18.1079L16.4993 17.5643L18.0943 19.5684H16.832L15.8895 18.334L14.9698 19.5684H13.7254Z"
                  fill="white"
                />
                <path
                  d="M18.2125 15.5606V19.5689H19.241V18.3031H20.2958C21.1884 18.3031 21.8649 17.8296 21.8649 16.9088C21.8649 16.1459 21.3343 15.5606 20.426 15.5606H18.2125ZM19.241 16.4671H20.3519C20.6402 16.4671 20.8463 16.6438 20.8463 16.9286C20.8463 17.1961 20.6413 17.39 20.3486 17.39H19.241V16.4671Z"
                  fill="white"
                />
                <path
                  d="M22.3007 15.5601V19.5685H23.3028V18.1444H23.7247L24.9262 19.5685H26.1508L24.8323 18.0917C25.3734 18.046 25.9316 17.5816 25.9316 16.8605C25.9316 16.017 25.2696 15.5601 24.5307 15.5601H22.3007ZM23.3028 16.4567H24.4482C24.723 16.4567 24.9229 16.6716 24.9229 16.8786C24.9229 17.1449 24.6639 17.3006 24.4631 17.3006H23.3028V16.4567Z"
                  fill="white"
                />
                <path
                  d="M30.085 19.5685V18.6982H32.0905C32.3873 18.6982 32.5157 18.5379 32.5157 18.362C32.5157 18.1935 32.3877 18.0231 32.0905 18.0231H31.1842C30.3965 18.0231 29.9578 17.5432 29.9578 16.8226C29.9578 16.1799 30.3595 15.5601 31.5301 15.5601H33.4816L33.0596 16.462H31.3719C31.0493 16.462 30.95 16.6313 30.95 16.7929C30.95 16.9591 31.0727 17.1423 31.3192 17.1423H32.2685C33.1467 17.1423 33.5277 17.6405 33.5277 18.2928C33.5277 18.9941 33.1031 19.5685 32.2207 19.5685H30.085Z"
                  fill="white"
                />
                <path
                  d="M33.7621 19.5685V18.6982H35.7676C36.0644 18.6982 36.1929 18.5379 36.1929 18.362C36.1929 18.1935 36.0648 18.0231 35.7676 18.0231H34.8614C34.0736 18.0231 33.6349 17.5432 33.6349 16.8226C33.6349 16.1799 34.0367 15.5601 35.2073 15.5601H37.1188L36.7368 16.462H35.049C34.7264 16.462 34.6271 16.6313 34.6271 16.7929C34.6271 16.9591 34.7498 17.1423 34.9963 17.1423H35.9456C36.8238 17.1423 37.2049 17.6405 37.2049 18.2928C37.2049 18.9941 36.7802 19.5685 35.8978 19.5685H33.7621Z"
                  fill="white"
                />
              </svg>
              <div
                class="css-j7qwjs"
              >
                <p
                  class="chakra-text css-1o3wn7d"
                >
                  American Express **** 8092
                </p>
                <p
                  class="chakra-text css-1q8knsm"
                >
                  Expires on 12/26
                </p>
              </div>
            </div>
          </span>
        </label>
      </div>
    </div>
  </div>
</div>
`;
